### PR TITLE
Fixes that the settings system crashes on an empty or missing settings file

### LINKED ISSUE #2073: Large amount of NullReferenceException when running StyleCop.Analyzers from MSBuild
We have recently updated our project from v1.0.0-rc1 to v1 and since then, we have been getting a large amount of NullReferenceException from our build server which is building the solution using MSBuild directly. We do not see those errors when using Quick Fixes or building our app in Visual Studio (even tough Run CA on build is enabled on the project). Here is the log I'm getting from MSBuild.

I think this is related to #1931 for which the fix was needed by our project. We also enabled SA0000 in the move over to the final release.

```
CSC : error AD0001: Analyzer 'StyleCop.Analyzers.OrderingRules.SA1203ConstantsMustAppearBeforeFields' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'. [d:\Jenkins\workspace\Departments\RB\ScheduAll.Reports\Dev\Build\src\RB.ScheduAll.Reports.Core\RB.ScheduAll.Reports.Core.csproj]
CSC : error AD0001: Analyzer 'StyleCop.Analyzers.MaintainabilityRules.SA1402FileMayOnlyContainASingleClass' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'. [d:\Jenkins\workspace\Departments\RB\ScheduAll.Reports\Dev\Build\src\RB.ScheduAll.Reports.Core\RB.ScheduAll.Reports.Core.csproj]
CSC : error AD0001: Analyzer 'StyleCop.Analyzers.OrderingRules.SA1208SystemUsingDirectivesMustBePlacedBeforeOtherUsingDirectives' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'. [d:\Jenkins\workspace\Departments\RB\ScheduAll.Reports\Dev\Build\src\RB.ScheduAll.Reports.Core\RB.ScheduAll.Reports.Core.csproj]
CSC : error AD0001: Analyzer 'StyleCop.Analyzers.OrderingRules.SA1214ReadonlyElementsMustAppearBeforeNonReadonlyElements' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'. [d:\Jenkins\workspace\Departments\RB\ScheduAll.Reports\Dev\Build\src\RB.ScheduAll.Reports.Core\RB.ScheduAll.Reports.Core.csproj]
CSC : error AD0001: Analyzer 'StyleCop.Analyzers.OrderingRules.SA1210UsingDirectivesMustBeOrderedAlphabeticallyByNamespace' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'. [d:\Jenkins\workspace\Departments\RB\ScheduAll.Reports\Dev\Build\src\RB.ScheduAll.Reports.Core\RB.ScheduAll.Reports.Core.csproj]
CSC : error AD0001: Analyzer 'StyleCop.Analyzers.LayoutRules.SA1518UseLineEndingsCorrectlyAtEndOfFile' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'. [d:\Jenkins\workspace\Departments\RB\ScheduAll.Reports\Dev\Build\src\RB.ScheduAll.Reports.Core\RB.ScheduAll.Reports.Core.csproj]
CSC : error AD0001: Analyzer 'StyleCop.Analyzers.DocumentationRules.SA1649FileNameMustMatchTypeName' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'. [d:\Jenkins\workspace\Departments\RB\ScheduAll.Reports\Dev\Build\src\RB.ScheduAll.Reports.Core\RB.ScheduAll.Reports.Core.csproj]
CSC : error AD0001: Analyzer 'StyleCop.Analyzers.NamingRules.SA1305FieldNamesMustNotUseHungarianNotation' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'. 
```

I am willing to provide a PR as a fix, but I will need further guidance on how to debug this use case.

### PR DESCRIPTION
**Pull request by pdelvo**
Fixes #2073.

### CODE REVIEW

**Comment by sharwell** [comment]
Can you verify that [SA0002](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA0002.md) gets reported in this scenario as well?

**Comment by pdelvo** [comment]
I added a test. I hope this solution is okay because otherwise I would probably have to do a lot of changes to the test system to test an empty string as the settings file. But this does not make any difference for the thing we try to test.

**Comment by sharwell** [comment]
:+1:

File: StyleCop.Analyzers/StyleCop.Analyzers/Settings/SettingsHelper.cs [hunk 1]
```diff
@@ -160,6 +166,12 @@ private static StyleCopSettings GetStyleCopSettings(ImmutableArray<AdditionalTex
                     {
                         SourceText additionalTextContent = additionalFile.GetText(cancellationToken);
                         var root = JsonConvert.DeserializeObject<SettingsFile>(additionalTextContent.ToString());
+
+                        if (root == null)
+                        {
+                            throw new JsonException($"Settings file at `{Path.GetFileName(additionalFile.Path)}` was missing or empty.");
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/Settings/SettingsHelper.cs]: :bulb: Use `'` instead of `` ` ``.

Comment (pdelvo) [inline StyleCop.Analyzers/StyleCop.Analyzers/Settings/SettingsHelper.cs]: Fixed

**Pull request merged**